api-analytics/service-worker.js

```diff
@@ -11,19 +11,30 @@ self.onactivate = function(event) {
   event.waitUntil(self.clients.claim());
 };
 
-// Fetch is as simply as log the request and passthrough.
-// Water clear thanks to the promise syntax!
 self.onfetch = function(event) {
-  event.respondWith(log(event.request).then(fetch));
+  event.respondWith(
+    // Log the request ...
+    log(event.request)
+    // .. and then actually perform it.
+    .then(fetch)
+  );
 };
 
 // Post basic information of the request to a backend for historical purposes.
 function log(request) {
-  var returnRequest = function() { return Promise.resolve(request); };
-  var data = { method: request.method, url: request.url };
+  var returnRequest = function() {
+    return request;
```

Comment:
I prefer the form that doesn't return the Promise for synchronous functions.
It's true that it can't appear at the beginning, but since it's synchronous you don't need to do `returnRequest().then`, you can just call it and get the return value in a local variable.
It can appear in the middle of the chain, for example try running this script in the console:
```JavaScript
var str = 'marco';
function returnStr() {
  return str;
}
Promise.resolve().then(returnStr).then(console.log.bind(console));
```